### PR TITLE
Users/mitokic/run model parallel bug fix

### PR DESCRIPTION
**Pull request by mitokic**
bug fixes from initial testing. Also implemented features from the "launch criteria" list.

### CODE REVIEW

File: R/forecast_models.R [hunk 1]
```diff
@@ -234,107 +233,114 @@ construct_forecast_models <- function(full_data_tbl,
...
+    cli::cli_h2("Running Combo: {combo_value}")
     
-    #filter on specific combo or all data
-    model_name_suffix <-  ifelse(combo_value=="All-Data","-all","") 
+    # Copy functions into global environment for azure batch
+    if(parallel_processing == "azure_batch") {
+      
+      global_env <- .GlobalEnv
+      
+      for(n in ls(global_env$azbatchenv$exportenv, all.names=TRUE)) assign(n, get(n, global_env$azbatchenv$exportenv), global_env)
```
Comment (AKannanMSFT) [inline R/forecast_models.R]: Optional: You may want to move this op out as a function call into ```azure_batch_parallel.R``` function

``` R
export_env_fully <- function(){
  global_env <- .GlobalEnv
  export_env <- global_env$azbatchenv$exportenv
      for(n in ls(export_env , all.names=TRUE)) assign(n, get(n, export_env), global_env)
}
```

File: R/forecast_models.R [hunk 2]
```diff
@@ -234,107 +233,114 @@ construct_forecast_models <- function(full_data_tbl,
...
+
     # create modeltime table to add single trained models to
     combined_models_recipe_1 <- modeltime::modeltime_table()
     combined_models_recipe_2 <- modeltime::modeltime_table()
-    
+
     # parallel processing
     if(run_model_parallel==TRUE & parallel_processing!="local_machine") {
-      
-      parallel_args <- parallel_init_func()
```
Comment (AKannanMSFT) [inline R/forecast_models.R:305]: Great Catch!

File: R/forecast_models.R [hunk 3]
```diff
@@ -400,79 +406,90 @@ construct_forecast_models <- function(full_data_tbl,
...
-    
+
     print(combined_models_recipe_1)
     print(combined_models_recipe_2)
     
+    cli::cli_h3("Refitting Individual Models")
+
     #create resamples for back testing forecasts and ensemble training data
+    
+    if(run_ensemble_models & (length(unique(combined_models_recipe_1$.model_desc))+length(unique(combined_models_recipe_2$.model_desc)))>1 & sum(grepl("-R", c(unique(combined_models_recipe_1$.model_desc), unique(combined_models_recipe_2$.model_desc)))) > 0) {
```
Comment (AKannanMSFT) [inline R/forecast_models.R:427]: Nit: Please leave a comment above this line on this condition

Comment (mitokic) [reply R/forecast_models.R:427]: Added comments in latest commit

File: R/forecast_models.R (hunk 1 above)
Comment (mitokic) [reply R/forecast_models.R]: tried adding function in latest commit, but ran into errors when exporting to azure global env. Kept as is for initial launch and is only called once, but could be moved to function for cleaner code in the future after initial launch.

Comment (AKannanMSFT) [reply R/forecast_models.R]: This change makes the code block readable. Which was the original intent of the comment. Thank you for making that change.

**Pull request merged**